exit.ts

```diff
@@ -116,7 +116,7 @@ export enum ExitCode {
 /**
  * A function that exits the app.
  */
-export type ExitFn = ((exitCode: number) => void) | typeof process.exit;
+export type ExitFn = (exitCode: number) => Promise<void>;
```

Comment:
This is an artifact from when I was calling shutdown in the exit function. But I prefer this interface, as it will be more flexible in the future.